tobiasHe/assignments/mergeIntervals/lc57/lc57.js

```diff
@@ -0,0 +1,111 @@
+// Problem Statement #
+
+// Given a list of non-overlapping intervals sorted by their start time, insert a given interval at the correct position and merge all necessary intervals to produce a list that has only mutually exclusive intervals.
+
+class Interval {
+	constructor(start, end) {
+		this.start = start;
+		this.end = end;
+	}
+
+	print_interval() {
+		process.stdout.write(`[${this.start}, ${this.end}]`);
+	}
+}
+
+const insert = function(intervals, new_interval) {
+	// TODO: Write your code here
+	// determine where the new interval will overlap
+	// since all the given intervals are not overlapping and sorted we can safely assume that if the start of new interval is greater than the end times
+	// then we can push the new interval after the ones that satisfy (new_interval.start > interval.end[...])
+	// find relevant intervals that the new interal affects it and insert it
+
+	let merged = [];
+	let i = 0;
+	let length = intervals.length;
+	//edge cases  if empty list of intervals just return newIntervals in the array
+	if (length < 1) {
+		merged.push(new_interval);
+		return merged;
+	}
+	//find the first place to insert our new_interval which is right before the interval it is overlapping with
+	while (i < length && intervals[i].end < new_interval.start) {
+		merged.push(intervals[i]);
+		i++;
+	}
+
+	//merging intervals
+	//all the first that is overlapping with second
+	while (i < length && intervals[i].start <= new_interval.end) {
+		new_interval[0] = Math.min(new_interval[0], intervals[i][0]);
+		new_interval[1] = Math.max(new_interval[1], intervals[i][1]);
+		i++;
+	}
+	merged.push(new_interval);
+
+	while (i < length) {
+		merged.push(intervals[i]);
+		i++;
+	}
+	return merged;
+};
```

Comment:
I tested this code and the incorrect result outputted. If you are unsure why, take a look at the solution and reverse engineer it.

Reply:
Oh crap, I forgot to save, I'll fix it!